src/coreclr/nativeaot/System.Private.DisabledReflection/src/Internal/Reflection/RuntimeTypeInfo.cs

```diff
@@ -202,6 +202,36 @@ protected override bool IsPrimitiveImpl()
             return RuntimeAugments.IsPrimitive(_typeHandle);
         }
 
+        public override bool IsAssignableFrom([NotNullWhen(true)] Type c)
+        {
+            if (c == null)
+                return false;
+
+            if (object.ReferenceEquals(c, this))
+                return true;
+
+            if (RuntimeAugments.IsGenericTypeDefinition(_typeHandle))
+                return false;
+
+            c = c.UnderlyingSystemType;
+
+            Type typeInfo = c;
+            RuntimeTypeInfo toTypeInfo = this;
+
+            if (typeInfo is not RuntimeType)
+                return false;  // Desktop compat: If typeInfo is null, or implemented by a different Reflection implementation, return "false."
+
+            RuntimeTypeInfo fromTypeInfo = (RuntimeTypeInfo)typeInfo;
+
+            RuntimeTypeHandle toTypeHandle = toTypeInfo._typeHandle;
+            RuntimeTypeHandle fromTypeHandle = fromTypeInfo._typeHandle;
```

Comment:
We need to do this for both type handles and we should throw when we find this because we can't decide castability based on runtime handles alone.

```suggestion
            c = c.UnderlyingSystemType;

            Type typeInfo = c;
            RuntimeTypeInfo toTypeInfo = this;

            if (typeInfo is not RuntimeType)
                return false;  // Desktop compat: If typeInfo is null, or implemented by a different Reflection implementation, return "false."

            RuntimeTypeInfo fromTypeInfo = (RuntimeTypeInfo)typeInfo;

            RuntimeTypeHandle toTypeHandle = toTypeInfo._typeHandle;
            RuntimeTypeHandle fromTypeHandle = fromTypeInfo._typeHandle;
            
            if (RuntimeAugments.IsGenericTypeDefinition(toTypeHandle) || RuntimeAugments.IsGenericTypeDefinition(fromTypeHandle))
                throw new NotSupportedException(SR.Reflection_Disabled);

```